lib/matplotlib/_layoutgrid.py

```diff
@@ -0,0 +1,607 @@
+
+    def edit_margin_min(self, todo, width, col=0):
+        """
+        Change the minimum size of the margin for one cell.
+
+        Parameters
+        ----------
+        todo : string (one of 'left', 'right', 'bottom', 'top')
+            margin to alter.
+
+        width : float
+            Minimum size of the margin .  If it is larger than the
+            existig minimum it updates the margin size. Fraction of
+            figure size.
+
+        col: int
+            Cell column or row to edit.
+        """
+
+        if width > self.margin_vals[todo][col]:
+            self.edit_margin(todo, width, col)
+
+    def edit_margins(self, todo, width):
+        """
+        Change the size of all the margin of all the cells in the layout grid.
+
+        Parameters
+        ----------
+        todo : string (one of 'left', 'right', 'bottom', 'top')
+            margin to alter.
+
+        width : float
+            Size to set the margins.  Fraction of figure size.
+        """
+
+        for i in range(len(self.margin_vals[todo])):
+            self.edit_margin(todo, width, i)
+
+    def edit_margins_min(self, todo, width):
+        """
+        Change the minimum size of all the margin of all
+        the cells in the layout grid.
+
+        Parameters
+        ----------
+        todo: string (one of 'left', 'right', 'bottom', 'top')
+            margin to alter.
+
+        width: float
+            Minimum size of the margin .  If it is larger than the
+            existig minimum it updates the margin size. Fraction of
+            figure size.
+        """
+
+        for i in range(len(self.margin_vals[todo])):
+            self.edit_margin_min(todo, width, i)
+
+    def edit_outer_margin_mins(self, margin, ss):
+        """
+        Edit all four margin minimums in one statement.
+
+        Parameters
+        ----------
+        margin: dict
+            size of margins in a dict with keys 'left', 'right', 'bottom',
+            'top'
+
+        ss: SubplotSpec
+            defines the subplotspec these margins should be applied to
+        """
+
+        self.edit_margin_min('left', margin['left'], ss.colspan.start)
+        self.edit_margin_min('leftcb', margin['leftcb'], ss.colspan.start)
+        self.edit_margin_min('right', margin['right'], ss.colspan.stop - 1)
+        self.edit_margin_min('rightcb', margin['rightcb'], ss.colspan.stop - 1)
+        # rows are from the top down:
+        self.edit_margin_min('top', margin['top'], ss.rowspan.start)
+        self.edit_margin_min('topcb', margin['topcb'], ss.rowspan.start)
+        self.edit_margin_min('bottom', margin['bottom'], ss.rowspan.stop - 1)
+        self.edit_margin_min('bottomcb', margin['bottomcb'],
+                             ss.rowspan.stop - 1)
+
+    def get_margins(self, todo, col):
+        """Return the margin at this position"""
+        return self.margin_vals[todo][col]
+
+    def get_outer_bbox(self, rows=[0], cols=[0]):
```

Comment:
Don't use mutable values as defaults.  I guess 0 as default is fine.  (Same comment throughout.)